breadbox/breadbox/api/downloads.py

```diff
@@ -67,7 +69,10 @@ def export_dataset(
     user: str = Depends(get_user),
     settings: Settings = Depends(get_settings),
 ):
-
+    try:
```

Comment:
I wonder if we could consolidate these calls.

Instead of calling utils.check_celery() before job submission, what do you think about putting it in get_task_status? 

That's when the worker being offline is a problem: the status will be reported as pending forever.

Reply:
Interesting! I noticed half of the functions using celery returned either `get_task_status()` or `cast_celery_task()` so I wasn't sure if I should have put it in `get_task_status` since not all functions calling a celery task calls it. Ideally, I would put them in celery's `delay()` function but patching that function felt too hacky…

I figured the fail fast approach would be better. I also considered creating a decorator or using fastapi's dependency injection `Depends()` but that would still modify each endpoint regardless